Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use Alpine-node as a base image because it is much smaller.
+#
+# Base image size for mhart/alpine-node:4 is 13 MB
+# Base image for standard node:4 is 253 MB.
+#
+# Our production deploy does not have any C/C++ library dependencies yet
+# so this is still easy to do.
+FROM mhart/alpine-node:4
```

Comment:
I actually don't believe there should be any code sharing between the Docker deploy and Ansiable.

If we were thinking of the docker deploys as being fully formed sub-operating systems that run more like a light-weight VM, then I agree. However, that sounds more and more like an anti-pattern. For example, I wouldn't ever want to use upstart to manage my process inside the docker container. Instead, I would want my process to be the equivalent of the OS init, and then have something monitor the `docker run` call and restart there.

As for how I stumbled on this image....I didn't actually do much.... just Google searched and checked to see if it was commonly used. Note that the mhart/alpine-node image is a layer on top of the official alpine linux install which just adds node. See https://github.com/mhart/alpine-node/blob/master/Dockerfile#L1.